src/test/scala/dataclass/CachedHashCodeTests.scala

```diff
@@ -0,0 +1,31 @@
+package dataclass
+
+import utest._
+
+object CachedHashCodeTests extends TestSuite {
+  // This is extremely crude naturally, but we typically get orders of magnitude difference:
+  //  for instance 26519 < 4382593
+  private def timed[T](warmup: => T)(t: => T): Long = {
+    warmup
+    val t0 = System.nanoTime()
+    t
+    System.nanoTime() - t0
+  }
+
+  val tests = Tests {
```

Comment:
About testing that, wouldn't something along those lines work and be more reliable?
```scala
class HashCodeCount(value: Int) {
  var hashCodeCount = 0
  override def hashCode(): Int = {
    hashCodeCount += 1
    value
  }
}

@data(cachedHashCode = true) class Cached(value: HashCodeCount)
@data class NotCached(value: HashCodeCount)

val cached = Cached(HashCodeCount(2))
val notCached = Cached(HashCodeCount(2))

assert(cached.value.hashCodeCount == 0)
assert(notCached.value.hashCodeCount == 0)

cached.hashCode
notCached.hashCode
assert(cached.value.hashCodeCount == 1)
assert(notCached.value.hashCodeCount == 1)

cached.hashCode
notCached.hashCode
assert(cached.value.hashCodeCount == 1)
assert(notCached.value.hashCodeCount == 2)
```

Reply:
Thanks, that's way better